autowiring/auto_signal.h

```diff
@@ -44,10 +61,24 @@ namespace autowiring {
       signal_node(std::function<void(Args...)>&& fn) :
         fn(std::move(fn))
       {}
-
+      
+      //Functions where the first argument is a signal_node<...> or base type are also ok.
+      signal_node(std::function<void(signal_node<Args...>*, Args...)>&& newFn) :
+      fn([this, newFn](Args... args){ newFn(this, args...); })
```

Comment:
If I did that, then I'd also have to add a Call Function method and support structure so that the correct one would be called for each node.